weave/graph_client.py

```diff
@@ -118,3 +164,95 @@ def feedback(self, feedback_id: str) -> typing.Optional[dict[str, typing.Any]]:
             if feedback_attrs["feedback_id"] == None:
                 return None
             return feedback_attrs
+
+    def ref_is_own(self, ref: typing.Optional[ref_base.Ref]) -> bool:
+        return isinstance(ref, artifact_wandb.WandbArtifactRef)
+
+    ##### Write API
+
+    def save_object(
+        self, obj: typing.Any, name: str, branch_name: str
+    ) -> artifact_wandb.WandbArtifactRef:
+        from . import storage
+
+        return storage._direct_publish(
+            obj,
+            name=name,
+            wb_entity_name=self.entity_name,
+            wb_project_name=self.project_name,
+            branch_name=branch_name,
+        )
+
+    def create_run(
+        self,
+        op_name: str,
+        parent: typing.Optional["RunKey"],
+        inputs: typing.Dict[str, typing.Any],
+        input_refs: list[artifact_wandb.WandbArtifactRef],
```

Comment:
seems a little odd that the input_refs are not derived from / directly included in the inputs themselves

Reply:
Yeah, this will keep changing.... lots of iterating here.